components/director/internal/domain/scenarioassignment/service.go

```diff
@@ -27,16 +27,25 @@ type ScenariosDefService interface {
 	GetAvailableScenarios(ctx context.Context, tenantID string) ([]string, error)
 }
 
-func NewService(repo Repository, scenarioDefSvc ScenariosDefService) *service {
+//go:generate mockery -name=AssignmentEngineService -output=automock -outpkg=automock -case=underscore
+type AssignmentEngineService interface {
+	EnsureScenarioAssignedToRuntimesMatchingSelector(in model.AutomaticScenarioAssignment) error
```

Comment:
minor: maybe those method names will be simpler:
```
	EnsureUpToDateLabels
	RemoveAssignedScenario
	RemoveAssignedScenarios
```

Reply:
or even we can have the only one `Remove` method and call it in the loop.

Reply:
> or even we can have the only one Remove method and call it in the loop.

I thought about it, but ideally I would like to do it on database level (with one query) or at least embed the loop in engine itself.